### PR TITLE
P1 radiation

### LINKED ISSUE #27725: Adding P1 Radiative Heat Transfer Model
### Motivation

P1 radiation heat transfer modeling is a good approximation for opaque heat transfer media. This approximation is particularly useful for radiation interacting with fluid flow. Currently, there is not a mantained P1 model for radiative heat transfer.

### Design

The P1 model should include:
- The radiation source and sink terms in the P1 approximation
- An implementation of the Marshak boundary conditions
- A material for computing the radiative heat transfer diffusion coefficient from the opacity and scattering cross sections

Also, the model should work with FV to be coupled to fluid flow.

### Impact

Adding P1 modeling if radiative heat transfer will enable several applications, among others:
- Bulk heat transfer in molten salt reactors and fluoride-cooled higher-temperature reactors
- Coupled fluid dynamics and radiative heat transfer in inertial confinement fusion

### PR DESCRIPTION
**Pull request by tanoret**
#closes #27725

### CODE REVIEW

**Review by simopier — CHANGES_REQUESTED** [review]
That looks pretty good to me (but should get another pair of eyes). I have a couple questions for clarification and a couple of minor suggestions. 
Thanks @tanoret!

File: modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/tests
Comment (lindsayad) [inline modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/tests:11]:
```suggestion
      detail = " in 1D."
```

File: modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/tests
Comment (lindsayad) [inline modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/tests:18]:
```suggestion
      detail = " in 2D."
```

File: modules/heat_transfer/test/tests/fvkernels/tests
Comment (lindsayad) [inline modules/heat_transfer/test/tests/fvkernels/tests:9]:
```suggestion
    requirement = "P1 Radiation heat transfer should match an anlytical solution in 1D."
```

File: modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md
Comment (tanoret) [reply modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md:17]: This is correct, $\sigma_{s, eff} = 3 \sigma_{tr}$. However, $\sigma_{tr}$ is the P1 transport cross section, which is defined as $\sigma_{tr} = \sigma_s (1 - <\mu>)$, where $<\mu>$ is the average scattering cosine angle of the scattering collision. Note that if the scattering is isotropic, then all output directions are equally probable and, hence, $<\mu> = \int_{-1}^1 P(\mu) \mu d \mu = \int_{-1}^1 1 \mu d \mu = 0$

Comment (tanoret) [reply modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md:17]: Added incropera as a reference.

File: modules/heat_transfer/doc/content/source/fvbcs/FVMarshakRadiativeBC.md
Comment (tanoret) [reply modules/heat_transfer/doc/content/source/fvbcs/FVMarshakRadiativeBC.md:11]: Thank you. Added modest as a reference.

File: modules/heat_transfer/doc/content/source/fvbcs/FVMarshakRadiativeBC.md
Comment (tanoret) [reply modules/heat_transfer/doc/content/source/fvbcs/FVMarshakRadiativeBC.md:17]: Thanks. Added.

File: modules/heat_transfer/doc/content/source/fvbcs/FVMarshakRadiativeBC.md
Comment (tanoret) [reply modules/heat_transfer/doc/content/source/fvbcs/FVMarshakRadiativeBC.md:19]: The diffusion coefficients should be in [m]. It is $\Gamma = \frac{1}{3 \kappa + \sigma_{s, eff}}$, where the macroscopic optical properties $\kappa$ and $\sigma_{s, eff}$ are in 1/m

File: modules/heat_transfer/doc/content/source/fvkernels/FVThermalRadiationSourceSink.md
Comment (tanoret) [reply modules/heat_transfer/doc/content/source/fvkernels/FVThermalRadiationSourceSink.md:17]: Thanks. Corrected

File: modules/heat_transfer/doc/content/source/fvkernels/FVThermalRadiationSourceSink.md
Comment (tanoret) [reply modules/heat_transfer/doc/content/source/fvkernels/FVThermalRadiationSourceSink.md:18]: Thanks. Corrected

File: modules/heat_transfer/include/fvbcs/FVMarshakRadiativeBC.h
Comment (tanoret) [reply modules/heat_transfer/include/fvbcs/FVMarshakRadiativeBC.h:28]: Thanks. Changed

File: modules/heat_transfer/include/fvkernels/FVThermalRadiationSourceSink.h
Comment (tanoret) [reply modules/heat_transfer/include/fvkernels/FVThermalRadiationSourceSink.h:16]: Thanks. Updated

File: modules/heat_transfer/include/fvkernels/FVThermalRadiationSourceSink.h
Comment (tanoret) [reply modules/heat_transfer/include/fvkernels/FVThermalRadiationSourceSink.h:28]: Thanks. Corrected.

File: modules/heat_transfer/src/fvbcs/FVMarshakRadiativeBC.C
Comment (tanoret) [reply modules/heat_transfer/src/fvbcs/FVMarshakRadiativeBC.C:34]: Thanks. Bad copy-paste indeed

File: modules/heat_transfer/src/fvbcs/FVMarshakRadiativeBC.C
Comment (tanoret) [reply modules/heat_transfer/src/fvbcs/FVMarshakRadiativeBC.C:35]: Thanks. Was a bad copy-paste from a previous class.

File: modules/heat_transfer/src/fvbcs/FVMarshakRadiativeBC.C
Comment (tanoret) [reply modules/heat_transfer/src/fvbcs/FVMarshakRadiativeBC.C:41]: Thanks. Changed

File: modules/heat_transfer/src/fvbcs/FVMarshakRadiativeBC.C
Comment (tanoret) [reply modules/heat_transfer/src/fvbcs/FVMarshakRadiativeBC.C:47]: Thanks. Changed

File: modules/heat_transfer/src/fvkernels/FVThermalRadiationSourceSink.C
Comment (tanoret) [reply modules/heat_transfer/src/fvkernels/FVThermalRadiationSourceSink.C:21]: Thanks. Changed.

File: modules/heat_transfer/test/tests/functormaterials/P1_radiation/P1_radiation_material_test.i
Comment (tanoret) [reply modules/heat_transfer/test/tests/functormaterials/P1_radiation/P1_radiation_material_test.i:22]: Thanks. Corrected the block above.

File: modules/heat_transfer/test/tests/functormaterials/P1_radiation/P1_radiation_material_test.i
Comment (tanoret) [reply modules/heat_transfer/test/tests/functormaterials/P1_radiation/P1_radiation_material_test.i:35]: Thanks. Removed AD from the non-AD material test.

File: modules/heat_transfer/test/tests/functormaterials/P1_radiation/tests
Comment (tanoret) [reply modules/heat_transfer/test/tests/functormaterials/P1_radiation/tests:8]: Thanks. Corrected

File: modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/rad_istothermal_medium_1d.i
Comment (tanoret) [reply modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/rad_istothermal_medium_1d.i:65]: Thanks. Changed.

File: modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/rad_istothermal_medium_2d.i
Comment (tanoret) [reply modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/rad_istothermal_medium_2d.i:66]: Thanks. Changed

File: modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/tests
Comment (tanoret) [reply modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/tests:11]: Thanks. Changed.

File: modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/tests
Comment (tanoret) [reply modules/heat_transfer/test/tests/fvbcs/fv_marshak_bc/tests:18]: Thanks. Changed.

File: modules/heat_transfer/test/tests/fvkernels/tests
Comment (tanoret) [reply modules/heat_transfer/test/tests/fvkernels/tests:9]: Thanks. Changed.

File: modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md
Comment (simopier) [reply modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md]: Oh OK, so when the medium is isotropic, then $\sigma_{s,eff}=3\sigma_{tr}=3\sigma_s$ (since $<\mu>=0$). 
I see that the warning is now gone, so I'm good with the current version of the documentation. 


Thanks for spelling it out for me!

**Review by simopier — CHANGES_REQUESTED** [review]
I still have one question about the proper value to use in an isotropic medium, but it looks good to me otherwise (pending test results)

File: modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md
Comment (tanoret) [reply modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md]: Yes, you are write. I will delete this warning

**Comment by lindsayad** [comment]
I will approve once you are good here @simopier

File: modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md
Comment (tanoret) [reply modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md]: Better now?

**Review by simopier — APPROVED** [review]
Looks good to me @lindsayad ! Thanks @tanoret!

File: modules/heat_transfer/doc/content/source/fvbcs/FVMarshakRadiativeBC.md
Comment (simopier) [reply modules/heat_transfer/doc/content/source/fvbcs/FVMarshakRadiativeBC.md]: Then the documentation in `RadiativeP1DiffusionCoefficientMaterial.md` needs to be fixed.

File: modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md
Comment (simopier) [inline modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md]:
```suggestion
- $\kappa$ is the opacity (SI units $m^{-1}$)
- $\sigma_{s, eff}$ is the P1 effective scattering cross section (SI units $m^{-1}$), e.g., $3 \sigma_{tr}$ with $\sigma_{tr}$ being the transport cross section.
```

**Review by simopier — CHANGES_REQUESTED** [review]
Actually, the documentation is not consistent for units, so I have one last change request.

File: modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md
Comment (simopier) [reply modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md]: Yes, thank you!

File: modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md
Comment (tanoret) [reply modules/heat_transfer/doc/content/source/functormaterials/RadiativeP1DiffusionCoefficientMaterial.md]: Thanks. Corrected

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
This looks more like transport-corrected P0 than P1 to me

**Pull request merged**
